File: 0647-palindromic-substrings/0647-palindromic-substrings.java
class Solution { // tc is O(nsquare) and sc is O(1) // expand center
    public int countSubstrings(String s) {
        int n = s.length();
        int count = 0;
        for(int i = 0;i<n;i++){
            count += expand(s,i,i); // odd
            count += expand(s,i,i+1); // even
        }
        return count;

        
    }
    private int expand(String s,int start,int end){
        int count = 0;
   
        while(start >= 0 && end < s.length() && s.charAt(start) == s.charAt(end)){
            count++;
            start--;
            end++;
        } 

        return count;
    }
}

File: 0647-palindromic-substrings/0647-palindromic-substrings.py
class Solution:
    # TC: O(n^2)
    # SC: O(1)
    # Expand around center
    def countSubstrings(self, s: str) -> int:
        n = len(s)
        count = 0

        for i in range(n):
            count += self.expand(s,i,i) # odd length
            count += self.expand(s,i,i+1) # even length
        return count
    
    def expand(self,s,start,end):
        count = 0
        while(start >= 0 and end < len(s) and s[start] == s[end]):
            count += 1
            start -= 1
            end += 1
        return count